mkosi/__init__.py

```diff
@@ -1589,7 +1589,7 @@ def configure_dracut(args: CommandLineArguments, root: str) -> None:
     # efivarfs must be present in order to GPT root discovery work
     if args.esp_partno is not None:
         with open(os.path.join(dracut_dir, "30-mkosi-efivarfs.conf"), "w") as f:
-            f.write('add_drivers+=" efivarfs "\n')
+            f.write('[[ $(modinfo -k "$KERNELVERSION" -F filename efivarfs 2>/dev/null) == /* ]] && drivers+=" efivarfs "\n')
```

Comment:
Might want to use the dracut `$kernel` variable here. Yeah, I know... lower case and the name is not clear.

```suggestion
            f.write('[[ $(modinfo -k "$kernel" -F filename efivarfs 2>/dev/null) == /* ]] && drivers+=" efivarfs "\n')
```